### PR TITLE
New feature - Editing physical storage

### PR DESCRIPTION
**Pull request by llivne**
This PR add new feature for physical-storages, the ability to edit an existing physical-storage. We can change the storage name and/or the storage credentials (management-ip, user and password)

image

image

image

links
-----
https://github.com/ManageIQ/manageiq-ui-classic/pull/7836
https://github.com/ManageIQ/manageiq-api/pull/1066

### CODE REVIEW

File: app/models/manageiq/providers/autosde/storage_manager/physical_storage.rb [hunk 1]
```diff
@@ -10,6 +13,17 @@ def raw_delete_physical_storage
     EmsRefresh.queue_refresh(ems)
   end
 
+  def raw_update_physical_storage(_options = {})
```
Comment (agrare) [inline app/models/manageiq/providers/autosde/storage_manager/physical_storage.rb:16]: Leading underscore indicates to linters to ignore a variable that isn't used later on, `_options` is used so better to have it just be `options = {}`

File: app/models/manageiq/providers/autosde/storage_manager/physical_storage.rb [hunk 2]
```diff
@@ -10,6 +13,17 @@ def raw_delete_physical_storage
...
+    update_details = ext_management_system.autosde_client.StorageSystemUpdate(
+      :name           => _options['name'],
+      :password       => _options['password'] || "",
+      :user           => _options['user'] || "",
+      :management_ip  => _options['management_ip'] || "",
+    )
+    ext_management_system.autosde_client.StorageSystemApi.storage_systems_pk_put(ems_ref, update_details)
+    EmsRefresh.queue_refresh(ems)
```
Comment (agrare) [inline app/models/manageiq/providers/autosde/storage_manager/physical_storage.rb:24]: (Follow-up) We should get targeted refresh working here so you don't have to queue a full for every one of these: https://www.manageiq.org/docs/guides/providers/targeted_refresh

File: app/models/manageiq/providers/autosde/storage_manager/physical_storage.rb (hunk 1 above)
Comment (llivne) [reply app/models/manageiq/providers/autosde/storage_manager/physical_storage.rb]: fixed

File: app/models/manageiq/providers/autosde/storage_manager/physical_storage.rb (hunk 2 above)
Comment (llivne) [reply app/models/manageiq/providers/autosde/storage_manager/physical_storage.rb:24]: @agrare do you meant we need to do refactoring to the entire autosde provider to use targeted refresh? if so I will open a new ticket for this action, and I prefer to push this PR with the full refresh if its ok with you.

Comment (agrare) [reply app/models/manageiq/providers/autosde/storage_manager/physical_storage.rb:24]: > and I prefer to push this PR with the full refresh if its ok with you.

Correct that's what I meant by follow-up

I wrote up a guide on how to do this in https://www.manageiq.org/docs/guides/providers/targeted_refresh

Comment (llivne) [reply app/models/manageiq/providers/autosde/storage_manager/physical_storage.rb:24]: I opened an issue for myself to add targeted_refresh for autosde provider. will be done in a separate PR in the short term

**Review by agrare — APPROVED** [review]

**Comment by ereztheodorou** [comment]
@miq-bot add_label lasker/yes?

**Comment by ereztheodorou** [comment]
@miq-bot rm_label lasker/yes?

**Comment by ereztheodorou** [comment]
@miq-bot add_label morphy/yes?

**Comment by ereztheodorou** [comment]
@miq-bot remove_label lasker/yes?

**Comment by Fryguy** [comment]
Backported to `morphy` in commit a76d05a46846efbc2ec28483e0374ac609c263ce.

```text
commit a76d05a46846efbc2ec28483e0374ac609c263ce
Author: Adam Grare <adam@grare.com>
Date:   Wed Sep 1 12:58:11 2021 -0400

    Merge pull request #76 from Autosde/edit_ph_storage
    
    New feature - Editing physical storage
    
    (cherry picked from commit 6183a50f27ff8cc70cdac9f8c21c7269ffab6294)
```

**Pull request merged**